### PR TITLE
[4.2] mod_breadcrumbs. Fix several issues with ld+json structured data

### PR DESCRIPTION
**Pull request by GHSVS-de**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/38229

Fixes also: Position is wrongly set if the menu pathway for example contains Heading menu items.

### Summary of Changes
- Add a helper method `getHome()` that can be used without redundant code elsewhere to get the home item.
- Only use `$app->getLanguage()` if needed.
- Pick the home item (`$homeCrumb`) in `mod_breadcrumbs.php` if `showHome` is disabled.
- `default.php`:
  - Use an independant counter for positions.
  - Always set an itemListElement for startpage even if `showHome` is disabled.
  - Only set `<SCRIPT ld.json>` if BraedCrumbList is populated.

### Testing Instructions
- Create a `mod_breadcrumbs` module. On all pages. Play around with the settings in the following.
- Build a test scenario with different menu items, menu item types and different menu deepness. Use also menu items with type Heading in between.

The visible breadcrumb hasn't changed with this pr, only the `ld+json` part.

#### Either:
In frontend click through your menus and check the source code of displayed codes.
Search for `ld+json` to find a `<script>` tag like

`<script type="application/ld+json">{"@context":"https:\/\/schema.org","@type":"BreadcrumbList","itemListElement":[...]}</script>`

#### Or:
You can also use Browser Addons like "JSON-LD Tester" for chromium browsers (Chrome, Vivaldi ...) which uses Google testing tools.

#### Both:
- Check that all menu items are present and represent the correct pathway of the current page.
- Check that Heading menu items are not.
- Check that parameters `position` go from 1 to x in +1 steps.
- Check that start page (home) is always at `position: 1`.

### Actual result BEFORE applying this Pull Request
- Empty and invalid `"itemListElement":[]` on startpage if `showHome` is disabled in mod_breadcrumbs.
- Wrong position steps if Heading menu items are present in menu.
- Missing home entry on deeper pages (?? don't remember).

### Expected result AFTER applying this Pull Request
No errors in structured data when testing.

### Additional
A legend is going around that the last element must not have a `@id` (= forbidden) if it's the current page, which is wrong (I've tested that on several pages that use other tools for ld+json). Because I 
 don't want endless discussions here I've left the old behavior in mod_breadcrumbs (no `@id` for last element if it's current page).

### CODE REVIEW

File: modules/mod_breadcrumbs/tmpl/default.php [hunk 1]
```diff
@@ -74,12 +74,28 @@
             'itemListElement' => []
     ];
 
+    // Use an independent counter for positions. E.g. if headline items in pathway.
```
Comment (GHSVS-de) [inline modules/mod_breadcrumbs/tmpl/default.php]:
```suggestion
    // Use an independent counter for positions. E.g. if Heading items in pathway.
```

File: modules/mod_breadcrumbs/tmpl/default.php [hunk 2]
```diff
@@ -74,12 +74,28 @@
             'itemListElement' => []
     ];
 
+    // Use an independent counter for positions. E.g. if Heading items in pathway.
+    $itemsCounter = 0;
+
+    // If showHome is disabled use the fallback $homeCrumb for startpage at first position.
+    if (isset($homeCrumb))
+    {
```
Comment (richard67) [inline modules/mod_breadcrumbs/tmpl/default.php]:
```suggestion
    if (isset($homeCrumb)) {
```
Please fix code style (we meanwhile use PSR-123 standard) to fix the PHPCS check in drone CI.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 8ac90ae320436f6e57c136ee41a0d21316c1c337<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38285">issues.joomla.org/tracker/joomla-cms/38285</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 8ac90ae320436f6e57c136ee41a0d21316c1c337<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38285">issues.joomla.org/tracker/joomla-cms/38285</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38285">issues.joomla.org/tracker/joomla-cms/38285</a>.</sub>

**Comment by roland-d** [comment]
Thank you

**Pull request merged**